slimta/policy/headers.py

```diff
@@ -132,4 +135,61 @@ def apply(self, envelope):
         envelope.prepend_header('Received', data)
 
 
+class AddDkimHeader(QueuePolicy):
+    """Adds a Domain Key Identified Mail header
+       will by default sign all the headers (except the ones marked
+       as SHOULD NOT SIGN as stated in dkimpy doc)
+       if this is not the last header added, the following ones
+       will not be signed.
+       :param dkim: Dict of dicts indexed by domain (example.com)
+                    - privkey: private key: PEM file loaded in a string
+                    - selector: selector setup in DNS for the domain
+                    - signature_algorithm: (default rsa-sha256)
+                    - include-headers: headers to sign (by default, all
+                    except the ones marked as SHOULD NOT SIGN see
+                    dkimpy doc)
+       Refs:
+       https://www.dkim.org
+       https://gathman.org/pydkim/
+       https://launchpad.net/dkimpy
+    """
+
+    def __init__(self, dkim):
+        self.dkim = dkim
+        self.logger = logging.getLogger(__name__)
+
+    def apply(self, envelope):
+        from dkim import dkim_sign, DKIMException
```

Comment:
Once you separate this class into its own file, this import can go up to the top.